// core-package/lib/src/auth/hasher/phpFunctions.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ord = exports.substr = exports.sixCharRandom = exports.md5 = exports.strpos = exports.strlen = void 0;
const crypto_1 = __importDefault(require("crypto"));
function strlen(str) {
    return str.length;
}
exports.strlen = strlen;
function strpos(string, find) {
    return string.indexOf(find);
}
exports.strpos = strpos;
function md5(string, raw) {
    const hash = crypto_1.default.createHash('md5').update(string, 'latin1');
    const hashDigestValue = raw ? 'latin1' : 'hex';
    return hash.digest(hashDigestValue);
}
exports.md5 = md5;
function sixCharRandom() {
    let text = '';
    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    for (let i = 0; i < 6; i++)
        text += possible.charAt(Math.floor(Math.random() * possible.length));
    return text;
}
exports.sixCharRandom = sixCharRandom;
function substr(string, start, count) {
    return string.substring(start, start + count);
}
exports.substr = substr;
function ord(input) {
    const r = input.charCodeAt(0);
    return r;
}
exports.ord = ord;
