// core-package/lib/tests/mailer/payload.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const payload_1 = require("../../src/mailer/payload");
const src_1 = require("../../src");
describe('MailerPayload builder', () => {
    test('create proper payload object', () => {
        const result = payload_1.buildRequestPayload({
            html: 'HTML',
            subject: 'SUBJECT',
            to: 'TO',
            customId: 'CUSTOM ID',
            monitoringCategory: src_1.MonitoringCategories.SHARE_VIDEO,
        }, { ACCOUNT_EMAIL: 'ACCOUNT EMAIL' });
        expect(result).toEqual({
            Messages: [
                {
                    CustomID: 'CUSTOM ID',
                    From: {
                        Email: 'ACCOUNT EMAIL',
                        Name: "L'équipe ADN",
                    },
                    HTMLPart: 'HTML',
                    MonitoringCategory: 'share_a_video_to_a_friend',
                    Subject: 'SUBJECT',
                    To: [
                        {
                            Email: 'TO',
                        },
                    ],
                },
            ],
        });
    });
});
