// core-package/lib/src/httpStatuses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpStatuses = void 0;
var HttpStatuses;
(function (HttpStatuses) {
    HttpStatuses[HttpStatuses["CONTINUE"] = 100] = "CONTINUE";
    HttpStatuses[HttpStatuses["SWITCHING_PROTOCOLS"] = 101] = "SWITCHING_PROTOCOLS";
    HttpStatuses[HttpStatuses["EARLY_HINTS"] = 102] = "EARLY_HINTS";
    HttpStatuses[HttpStatuses["OK"] = 200] = "OK";
    HttpStatuses[HttpStatuses["CREATED"] = 201] = "CREATED";
    HttpStatuses[HttpStatuses["ACCEPTED"] = 202] = "ACCEPTED";
    HttpStatuses[HttpStatuses["NON_NON_AUTHORITATIVE_INFORMATION"] = 203] = "NON_NON_AUTHORITATIVE_INFORMATION";
    HttpStatuses[HttpStatuses["NO_CONTENT"] = 204] = "NO_CONTENT";
    HttpStatuses[HttpStatuses["RESET_CONTENT"] = 205] = "RESET_CONTENT";
    HttpStatuses[HttpStatuses["PARTIAL_CONTENT"] = 206] = "PARTIAL_CONTENT";
    HttpStatuses[HttpStatuses["MULTIPLE_CHOICES"] = 300] = "MULTIPLE_CHOICES";
    HttpStatuses[HttpStatuses["MOVED_PERMANENTLY"] = 301] = "MOVED_PERMANENTLY";
    HttpStatuses[HttpStatuses["FOUND"] = 302] = "FOUND";
    HttpStatuses[HttpStatuses["SEE_OTHER"] = 303] = "SEE_OTHER";
    HttpStatuses[HttpStatuses["NOT_MODIFIED"] = 304] = "NOT_MODIFIED";
    HttpStatuses[HttpStatuses["TEMPORARY_REDIRECT"] = 307] = "TEMPORARY_REDIRECT";
    HttpStatuses[HttpStatuses["PERMANENT_REDIRECT"] = 308] = "PERMANENT_REDIRECT";
    HttpStatuses[HttpStatuses["BAD_REQUEST"] = 400] = "BAD_REQUEST";
    HttpStatuses[HttpStatuses["UNAUTHORIZED"] = 401] = "UNAUTHORIZED";
    HttpStatuses[HttpStatuses["PAYMENT_REQUIRED"] = 402] = "PAYMENT_REQUIRED";
    HttpStatuses[HttpStatuses["FORBIDDEN"] = 403] = "FORBIDDEN";
    HttpStatuses[HttpStatuses["NOT_FOUND"] = 404] = "NOT_FOUND";
    HttpStatuses[HttpStatuses["METHOD_NOT_ALLOWED"] = 405] = "METHOD_NOT_ALLOWED";
    HttpStatuses[HttpStatuses["NOT_ACCEPTABLE"] = 406] = "NOT_ACCEPTABLE";
    HttpStatuses[HttpStatuses["PROXY_AUTHENTICATION_REQUIRED"] = 407] = "PROXY_AUTHENTICATION_REQUIRED";
    HttpStatuses[HttpStatuses["REQUEST_TIMEOUT"] = 408] = "REQUEST_TIMEOUT";
    HttpStatuses[HttpStatuses["CONFLICT"] = 409] = "CONFLICT";
    HttpStatuses[HttpStatuses["GONE"] = 410] = "GONE";
    HttpStatuses[HttpStatuses["LENGTH_REQUIRED"] = 411] = "LENGTH_REQUIRED";
    HttpStatuses[HttpStatuses["PRECONDITION_FAILED"] = 412] = "PRECONDITION_FAILED";
    HttpStatuses[HttpStatuses["PAYLOAD_TOO_LARGE"] = 413] = "PAYLOAD_TOO_LARGE";
    HttpStatuses[HttpStatuses["URI_TOO_LONG"] = 414] = "URI_TOO_LONG";
    HttpStatuses[HttpStatuses["UNSUPPORTED_MEDIA_TYPE"] = 415] = "UNSUPPORTED_MEDIA_TYPE";
    HttpStatuses[HttpStatuses["RANGE_NOT_SATISFIABLE"] = 416] = "RANGE_NOT_SATISFIABLE";
    HttpStatuses[HttpStatuses["EXPECTATION_FAILED"] = 417] = "EXPECTATION_FAILED";
    HttpStatuses[HttpStatuses["IM_A_TEAPOT"] = 418] = "IM_A_TEAPOT";
    HttpStatuses[HttpStatuses["UNPROCESSABLE_ENTITY"] = 422] = "UNPROCESSABLE_ENTITY";
    HttpStatuses[HttpStatuses["TOO_EARLY"] = 425] = "TOO_EARLY";
    HttpStatuses[HttpStatuses["UPGRADE_REQUIRED"] = 426] = "UPGRADE_REQUIRED";
    HttpStatuses[HttpStatuses["PRECONDITION_REQUIRED"] = 428] = "PRECONDITION_REQUIRED";
    HttpStatuses[HttpStatuses["TOO_MANY_REQUESTS"] = 429] = "TOO_MANY_REQUESTS";
    HttpStatuses[HttpStatuses["REQUEST_HEADER_FIELDS_TOO_LARGE"] = 431] = "REQUEST_HEADER_FIELDS_TOO_LARGE";
    HttpStatuses[HttpStatuses["UNAVAILABLE_FOR_LEGAL_REASONS"] = 451] = "UNAVAILABLE_FOR_LEGAL_REASONS";
    HttpStatuses[HttpStatuses["INTERNAL_SERVER_ERROR"] = 500] = "INTERNAL_SERVER_ERROR";
    HttpStatuses[HttpStatuses["NOT_IMPLEMENTED"] = 501] = "NOT_IMPLEMENTED";
    HttpStatuses[HttpStatuses["BAD_GATEWAY"] = 502] = "BAD_GATEWAY";
    HttpStatuses[HttpStatuses["SERVICE_UNAVAILABLE"] = 503] = "SERVICE_UNAVAILABLE";
    HttpStatuses[HttpStatuses["GATEWAY_TIMEOUT"] = 504] = "GATEWAY_TIMEOUT";
    HttpStatuses[HttpStatuses["HTTP_VERSION_NOT_SUPPORTED"] = 505] = "HTTP_VERSION_NOT_SUPPORTED";
    HttpStatuses[HttpStatuses["VARIANT_ALSO_NEGOTIATES"] = 506] = "VARIANT_ALSO_NEGOTIATES";
    HttpStatuses[HttpStatuses["INSUFFICIENT_STORAGE"] = 507] = "INSUFFICIENT_STORAGE";
    HttpStatuses[HttpStatuses["LOOP_DETECTED"] = 508] = "LOOP_DETECTED";
    HttpStatuses[HttpStatuses["NOT_EXTENDED"] = 510] = "NOT_EXTENDED";
    HttpStatuses[HttpStatuses["NETWORK_AUTHENTICATION_REQUIRED"] = 511] = "NETWORK_AUTHENTICATION_REQUIRED";
})(HttpStatuses = exports.HttpStatuses || (exports.HttpStatuses = {}));
