// core-package/lib/src/requester.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.requester = exports.RequesterMethods = void 0;
const axios_1 = __importDefault(require("axios"));
var RequesterMethods;
(function (RequesterMethods) {
    RequesterMethods["GET"] = "get";
    RequesterMethods["POST"] = "post";
    RequesterMethods["DELETE"] = "delete";
    RequesterMethods["HEAD"] = "head";
    RequesterMethods["OPTIONS"] = "options";
    RequesterMethods["PUT"] = "put";
    RequesterMethods["PATCH"] = "patch";
})(RequesterMethods = exports.RequesterMethods || (exports.RequesterMethods = {}));
function requester({ url, method, data, headers, params, timeout, }) {
    return __awaiter(this, void 0, void 0, function* () {
        const response = yield axios_1.default({
            url,
            method,
            data,
            headers,
            params,
            timeout,
        });
        return response.data;
    });
}
exports.requester = requester;
